### PR TITLE
when an email is picked, transition to email_chosen which itself decides...

### PR DESCRIPTION
**Pull request by lloyd**
... whether authenticaiton is neccesary - issue #2834

### CODE REVIEW

File: resources/static/dialog/js/modules/pick_email.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ BrowserID.Modules.PickEmail = (function() {
         else if ("transition_no_password" === info.state)
           self.close("transition_no_password", info);
         else if ("secondary" === info.type)
```
Comment (shane-tomlinson) [inline resources/static/dialog/js/modules/pick_email.js]: There should be tests for each of those cases.

**Comment by shane-tomlinson** [comment]
@lloyd - I have added tests and added some {}, would you mind re-reviewing to make sure I didn't change any logic?

File: resources/static/dialog/js/modules/pick_email.js [hunk 2]
```diff
@@ -61,15 +61,22 @@ BrowserID.Modules.PickEmail = (function() {
     var record = checkEmail.call(self, email);
     if (!! record) {
       dialogHelpers.refreshEmailInfo.call(self, email, function (info) {
-	record = checkEmail.call(self, email);
-        if (record.cert)
+        record = checkEmail.call(self, email);
```
Comment (shane-tomlinson) [inline resources/static/dialog/js/modules/pick_email.js:64]: Does record need to be fetched both here and on line 61?

Comment (lloyd) [inline resources/static/dialog/js/modules/pick_email.js:64]: sure don't think so.  Only if refreshEmailInfo can change the record?

**Pull request merged**